src/pages/Auth/Signup.jsx

```diff
@@ -0,0 +1,162 @@
+import React, { useEffect, useRef, useState } from "react";
+import "./Auth.css";
+import VisibilityOutlinedIcon from "@mui/icons-material/VisibilityOutlined";
+import VisibilityOffOutlinedIcon from "@mui/icons-material/VisibilityOffOutlined";
+import { Link } from "react-router-dom";
+
+const Signup = () => {
+  const [eye, setEye] = useState(true);
+  const [fName, setFName] = useState("");
+  const [lName, setLName] = useState("");
+  const [email, setEmail] = useState("");
+  const [password, setPassword] = useState("");
```

Comment:
Although there is nothing with using states here, you can just pass the ref.current.value in the payload on click of the submit button. This can help you avoid declaring state variables for each input.